core/common/src/main/java/alluxio/security/authorization/AccessControlList.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authorization;
+
+import alluxio.proto.journal.File;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Access control list for a file or directory.
+ */
+public final class AccessControlList {
+  /**
+   * Initial capacity for {@link #mUserActions} and {@link #mGroupActions}.
+   * Most of the time, only owning user and owning group exists in {@link #mUserActions} and
+   * {@link #mGroupActions}.
+   */
+  private static final int ACTIONS_MAP_INITIAL_CAPACITY = 1;
+  /** Initial load factor. */
+  private static final int ACTIONS_MAP_INITIAL_LOAD_FACTOR = 1;
+  /** Key representing owning user in {@link #mUserActions}. */
+  private static final String OWNING_USER_KEY = "";
+  /** Key representing owning group in {@link #mGroupActions}. */
+  private static final String OWNING_GROUP_KEY = "";
+
+  private String mOwningUser;
+  private String mOwningGroup;
+  private Map<String, AclActions> mUserActions;
+  private Map<String, AclActions> mGroupActions;
+  private AclActions mMaskActions;
+  private AclActions mOtherActions;
+
+  /**
+   * Creates a new instance where owning user and owning group are initialized to empty strings,
+   * and no action is permitted for any user or group.
+   */
+  public AccessControlList() {
+    mOwningUser = "";
+    mOwningGroup = "";
+    mUserActions = new HashMap<>(ACTIONS_MAP_INITIAL_CAPACITY, ACTIONS_MAP_INITIAL_LOAD_FACTOR);
+    mUserActions.put(OWNING_USER_KEY, new AclActions());
+    mGroupActions = new HashMap<>(ACTIONS_MAP_INITIAL_CAPACITY, ACTIONS_MAP_INITIAL_LOAD_FACTOR);
+    mGroupActions.put(OWNING_GROUP_KEY, new AclActions());
+    mMaskActions = new AclActions();
```

Comment:
Would it make sense to have a `static` default `AclActions` object, and then copy on write? This could save space in the common case.

This could be left as a TODO.

Reply:
added a TODO